ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Update.scala

```diff
@@ -71,17 +71,54 @@ object Update {
     * @param recordTime
     *   The ledger-provided timestamp at which the party was allocated.
     *
+    * @param submissionId
+    *   The submissionId of the command which requested party to be added.
+    *
     */
+  // TODO BH: add isLocal as was on async event
   final case class PartyAddedToParticipant(
       party: Party,
       displayName: String,
       participantId: ParticipantId,
-      recordTime: Timestamp)
+      recordTime: Timestamp,
+      submissionId: SubmissionId)
       extends Update {
     override def description: String =
       s"Add party '$party' to participant"
   }
 
+  /** Signal that the party allocation request has been Rejected.
+    *
+    * Initially this will be visible to all participants in the current open world,
+    * with a possible need to revisit as part of the per-party package visibility work
+    * https://github.com/digital-asset/daml/issues/311.
+    *
+    * @param participantId
```

Comment:
The order here seems to be out of line with the actual parameter ordering -- you might want to fix that.